feat: add attendee to event

Sofiag8 opened the pull request:
## Conference App - Backend

## Issue ticket number and link



## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. -->

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->

Please check the type of change your PR introduces:

- [x] :sparkles: Feature
- [ ] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [x] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [x] :white_check_mark: Unit testing
- [x] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

The endpoint existing at v1 about adding an user to an event is not working correctly and not good implemented

## What is the new behavior?

<!-- Please describe the behavior or changes that are being added by this PR. -->

Now a v2 we will have the new endpoint for adding an user to an event with necessary validations 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

FelipeLosada1996 commented on src/modules/events/event.controller.ts:
```diff
@@ -38,4 +40,12 @@ export class EventController {
+  @Post('/:eventId/attendee/:userId')
+  @HttpCode(200)
+  async addAttendeeToEvent(@Param() params: AddAttendeeToEventParams , @Body() attendeeData: AddAttendeeToEventDto): Promise<EventResponse> {
+    this.logger.log('Add attendee to event')
```
In what help us that log? we can add some extra data to find what can break the endpoint or if we need what gonna be created

FelipeLosada1996 commented on src/modules/events/event.service.ts:
```diff
@@ -41,4 +44,31 @@ export class EventService {
+  }
+
+  private validateUserAttendingEvent(event: Event, eventId: ObjectId, userId: ObjectId) {
+    this.logger.log('Validating event attendees to avoid duplicated attendees...')
```
What exactly do this log? is help us to something?

Sofiag8 commented:
@FelipeLosada1996  I fixed the master conflicts, and also merge changes with the previous created mock files, I cleaned tests files avoiding repeated mocked code.

FelipeLosada1996 commented on src/modules/events/event.controller.ts:
```diff
@@ -50,8 +53,16 @@ export class EventController {
   @Get('/:eventId')
   @HttpCode(200)
-  async getById(@Param('eventId') eventId: string): Promise<EventResponse> {
+  async getById(@Param('eventId') eventId: Types.ObjectId): Promise<EventResponse> {
```
This is not a good practice in the controller, check  https://stackoverflow.com/questions/71597229/nestjs-validate-param-format-of-mongo-objectid

FelipeLosada1996 commented on src/modules/events/event.controller.spec.ts:
```diff
@@ -71,43 +53,42 @@ describe('EventController', () => {
+      jest.spyOn(service, 'getById').mockResolvedValue(EVENT_RESPONSE_MOCK)
 
-      const result = await controller.getById(id)
+      const result = await controller.getById(Object(id))
```
What is the difference with `Object(EVENT_ID_MOCK)`?

Sofiag8 replied:
done

Sofiag8 commented on src/modules/events/event.controller.spec.ts:
```diff
@@ -71,43 +53,43 @@ describe('EventController', () => {
+      const response: EventResponse = EVENT_RESPONSE_MOCK
       jest.spyOn(service, 'getById').mockResolvedValue(response)
 
       const result = await controller.getById(id)
```
done
